include/votca/tools/calculator.h

```diff
@@ -77,20 +77,40 @@ class Calculator {
    * @param output stream
    */
   void DisplayOptions(std::ostream &out);
+
+  /**
+   * \brief Loads default options stored in VOTCASHARE
+   */
+  void LoadDefaults(std::string package = "tools");
+
   /**
    * \brief Updates options with default options stored in VOTCASHARE
    *
    * If a value is not given or tag is not present and at the same time
    * a default value exists in the corresponding XML file in VOTCASHARE
    * a tag is created and/or a default value is assigned to it
    */
-  void UpdateWithDefaults(Property &options, std::string package = "tools");
+  void UpdateWithUserOptions(const Property &options);
+
+  /**
+   * \brief Load the default options and merge them with the user input
+   *
+   * Defaults are overwritten with user input
+   */
+  void LoadDefaultsAndUpdateWithUserOptions(std::string package,
+                                            const Property &options) {
+    LoadDefaults(package);
+    UpdateWithUserOptions(options);
+  }
 
  protected:
   Index _nThreads;
   bool _maverick;
 
-  void AddDefaults(Property &p, const Property &defaults);
+  void AddDefaults(const Property &p, Property &defaults);
+  static std::string GetVotcaShare();
+
+  Property _options;  // the finally used options
```

Comment:
why does the calculator have to have this?

Reply:
Since we are using internal defaults that are amended with the user input during the initialization, we need to have a stateful variable to store all these settings for the calculators.

Reply:
Is a local copy then not enough?

Reply:
I mean the calculator loads the default options and then loads the useroptions if they exist. The result of that is parsed by the initialize method, which used to have a `Property& options` input. 

I do not see why the calculator has to store the whole property for its lifetime.

Reply:
especially because it is only used once in the beginning.

Reply:
That is true, let me change it